pos-print/src/main/java/com/clouway/pos/print/adapter/db/BackgroundReceiptPrinter.kt

```diff
@@ -0,0 +1,12 @@
+package com.clouway.pos.print.adapter.db
+
+import com.mongodb.client.MongoCollection
+import org.bson.Document
+
+/**
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+interface BackgroundReceiptPrinter {
+  fun run()
+  fun printReceipts(collection: MongoCollection<Document>)
```

Comment:
BackgroundReceiptPrinter interface gets coupled to mongodb which doesn't sound good. It would be better db to be passed to the constructor.

Reply:
I will implement an interface to abstract the receipt queue from the background thread and that should clear up these dependencies.